CMakeLists.txt

```diff
@@ -108,6 +108,12 @@ install(TARGETS pennylane_lightning
         LIBRARY DESTINATION lib
         ARCHIVE DESTINATION lib
         INCLUDES DESTINATION include
+        PUBLIC_HEADER DESTINATION include
+)
+
+install(DIRECTORY 
```

Comment:
I don't think I saw any occurrence of  `cmake --install ./build` in our workflows.
I like the idea of using such a feature in some of our workflows to test and serve as examples.
Should we also change the [README.rst](https://github.com/PennyLaneAI/pennylane-lightning#readme)?

Reply:
I think while that's a good plan, I'd like to test the waters first with Spack, and see if this is something useful. If so, we can support it through CLI installations and document it as such. I expect we will be updating this based on end-user feedback so rather than moving a lot, we can leave as is for now.